core/domain/user_services_test.py

```diff
@@ -164,13 +164,21 @@ def test_set_invalid_usernames(self):
             'AbcOppiaSuggestionBotXyz', 'AAAOPPIASuggestionBotBBB',
             'xyzOppia', 'oppiaXyz', 'abcOppiaXyz']
         for username in bad_usernames:
-            with self.assertRaises(utils.ValidationError):
+            with self.assertRaisesRegexp(
+                utils.ValidationError,
+                r'(Usernames can only have alphanumeric characters.)|'
```

Comment:
Change `bad_usernames` to `bad_usernames_with_expected_error_message` and create a list of tuple (usernmae, error_message), loop through the ist and test specific expected error for each username!